.github/workflows/s7-plc.yml

```
@@ -42,5 +42,42 @@ jobs:
           go_version: ${{ env.GO_VERSION }}
       - name: Install Ginkgo
         run: go install github.com/onsi/ginkgo/v2/ginkgo@v2.19.0
+      - name: Install Tcping
+        run: go install github.com/cloverstd/tcping@latest
+
+      - name: Check S7 port availability
+        id: check_s7
+        run: |
+          set +x
+          ENDPOINT="${{ secrets.TEST_S7_TCPDEVICE }}"
+          if tcping "$ENDPOINT" | grep -qi "Connected"; then
+            echo "available=true" >> "$GITHUB_OUTPUT"
+          else
+            echo "available=false" >> "$GITHUB_OUTPUT"
+          fi
+
+      - name: Check S7-fallback port availability
+        id: check_s7_fallback
```

An optimal algorithm  could be add an if condition in this step (refer this https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/using-conditions-to-control-job-execution#overview) 

```
if: ${{ steps.check_s7.outputs.available}}
```

and set the $ENDPOINT to the fallback url. 

By this optimisation, this step won't run if the previous step is available and the step after this can be removed completely as the Endpoint is already set. 
This optimisation can be done in the other file too where you did a similar change.

---

I have remodeled the workflow a bit regarding your comments.
We now use if-statements if the `main-device` is available, but it's not used for the testing-step since we want to exit & fail there if they're both not available.
On top of that we directly set the env-variable, so that we can check on it later on.